include/libcgroup/groups.h

```diff
@@ -675,6 +675,12 @@ struct cgroup_controller *cgroup_get_controller_by_index(struct cgroup *cgroup,
  */
 char *cgroup_get_controller_name(struct cgroup_controller *controller);
 
+/**
+ * Return true if cgroup setup mode is cgroup v2 (unified), else
+ * returns false.
+ */
+bool is_unified_mode(void);
```

Comment:
nitpick - we probably should remain consistent with previous APIs and use the `cgroup_` prefix.  How does `cgroup_is_unified_mode()` sound?

Reply:
Silly question - should we just expose `cgroup_setup_mode()` and give the user even more power.  I'm not sure where you're going with this function, so I could be way off base.

Reply:
@drakenclimber Thanks for testing and reviewing the patch.  `cgroup_is_unified_mode()` sounds better, as it's consistent with the other APIs and I will add test cases around it.

> Silly question - should we just expose cgroup_setup_mode() and give the user even more power? I'm not sure where you're going with this function, so I could be way off base.

The idea was to export `is_unified_mode()` as a helper to the users, so they can quickly check if the setup is v1 (legacy/hybrid) or v2.  If it returns true it is v2 or v1 (legacy or hybrid) otherwise.  Exposing `cgroup_setup_mode()` could definitely give more power to the user, but that would also mean exposing `enum cg_setup_mode_t`. 

The next question is, why not have `is_legacy_mode()` and `is_hybrid_mode()` functions?

Reply:
`enum cg_setup_mode_t` is already exported, will spin up the next version will `cgroup_setup_mode()` and other helpers.

Reply:
I don't have a preference either way.  If you think `cgroup_is_unified_mode()` is easier/better for the user, then I'm cool with that.  If we think giving them the full `cg_setup_mode_t` enum is easier, that's fine, too.